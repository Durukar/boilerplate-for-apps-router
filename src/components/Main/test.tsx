import { render, screen } from '@testing-library/react'

import { Main } from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // Renderiza o component
    const { container } = render(<Main />)

    // Busca o elemento e verifica a existência
    expect(
      screen.getByRole('heading', { name: /React Avançado/i })
    ).toBeInTheDocument()

    // Gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    // Renderiza o component
    const { container } = render(<Main />)

    // Verifica se a cor do background está correta
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
