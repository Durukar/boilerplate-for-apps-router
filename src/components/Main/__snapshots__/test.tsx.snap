// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-bdfCDU esWOmd"
>
  <h1>
    Casa da Mae Sylvannas
  </h1>
</main>
`;
